src/chips/safe_accumulator.rs

```diff
@@ -0,0 +1,211 @@
+use num_bigint::BigUint;
+use std::fmt::Debug;
+
+use super::is_zero::{IsZeroChip, IsZeroConfig};
+use super::utils::{decompose_bigInt_to_ubits, fp_to_big_uint, value_fp_to_big_uint, range_check_vec};
+use halo2_proofs::{circuit::*, halo2curves::pasta::Fp, plonk::*, poly::Rotation};
```

Comment:
For all the chips, I suggest keeping them generic to the Field. The Field should only be defined when the circuit is instantiated. For example:

- here is a [Chip definition](https://github.com/summa-dev/circuits-halo2/blob/master/src/chips/merkle_sum_tree.rs#L12) generic to the Field element.
- here is the [Circuit Instantiation](https://github.com/summa-dev/circuits-halo2/blob/master/src/circuits/merkle_sum_tree.rs#L132) that defines the Field

Reply:
Forget about this comment, I actually realized that working with generic fields is not worth it. [I removed the usage of generic type `FieldExt` in the merkle sum tree chip](https://github.com/summa-dev/summa-solvency/commit/dfc5d7e368c2aacff371845712230cd0c646d065). The only thing here that needs to be addressed is replacing the field with `bn256`